bmap-rs/src/main.rs

```diff
@@ -1,19 +1,22 @@
 use anyhow::{anyhow, bail, Context, Result};
-use bmap::{Bmap, Discarder, SeekForward};
+use bmap::{AsyncDiscarder, Bmap, Discarder, SeekForward};
 use clap::Parser;
 use flate2::read::GzDecoder;
+use futures::TryStreamExt;
 use indicatif::{ProgressBar, ProgressState, ProgressStyle};
 use nix::unistd::ftruncate;
+use reqwest::{Response, Url};
 use std::ffi::OsStr;
 use std::fmt::Write;
 use std::fs::File;
 use std::io::Read;
 use std::os::unix::io::AsRawFd;
 use std::path::{Path, PathBuf};
+use tokio_util::compat::TokioAsyncReadCompatExt;
 
 #[derive(Parser, Debug)]
 struct Copy {
-    image: PathBuf,
+    image: String,
```

Comment:
Would having `image: Optional<PathBuf>` and `image_url: Optional<Url>` make it less awkward ?

Reply:
I'm not sure how that's going to work with the clap parser but we can try

Reply:
From some googling I found https://github.com/clap-rs/clap/issues/2437; may be better to let clap do the URL/PATH parsing ?

Reply:
I've been suggested that in order to have a more elegant parsing the way would be to have a custom parser. I think maybe the scope of that exceed this PR and deserve it's own Issue.